### PR TITLE
Upgrade rest-common to 2.0.0-RC

### LINKED ISSUE #864: Model cannot be used without spring-web
As [collection API classes](https://github.com/gooddata/gooddata-rest-common/tree/master/src/main/java/com/gooddata/collections) depend on spring-web we cannot use SDK model modul without spring-web dependency even with #711 resolved

### PR DESCRIPTION
**Pull request by jimirocks**
Resolves #864

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-java/src/main/java/com/gooddata/sdk/service/project/ProjectService.java [hunk 1]
```diff
@@ -62,26 +89,13 @@ public ProjectService(final RestTemplate restTemplate, final AccountService acco
...
-    @Deprecated
-    public Collection<Project> getProjects() {
-        return listProjects().collectAll();
-    }
-
-    /**
-     * Get first page of paged list of projects that current user has access to.
+     * Get browser of projects that current user has access to.
      *
      * @return MultiPageList list of found projects or empty list
```
Comment (peter-plochan) [inline gooddata-java/src/main/java/com/gooddata/sdk/service/project/ProjectService.java]: Change return value in JavaDoc

File: gooddata-java/src/main/java/com/gooddata/sdk/service/project/ProjectService.java [hunk 2]
```diff
@@ -62,26 +89,13 @@ public ProjectService(final RestTemplate restTemplate, final AccountService acco
     }
 
     /**
-     * Get all projects current user has access to.
-     *
-     * @return collection of all projects current user has access to
-     * @throws com.gooddata.GoodDataException when projects can't be accessed
-     * @deprecated use {@link #listProjects()} or {@link #listProjects(Page)} instead.
-     * Deprecated since version 3.0.0. Will be removed in one of future versions.
```
Comment (liry) [inline gooddata-java/src/main/java/com/gooddata/sdk/service/project/ProjectService.java]: This was meant to keep deprecated in 3.0.0. Is there any problem with that?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-java/src/main/java/com/gooddata/sdk/service/project/ProjectService.java (hunk 2 above)
Comment (jimirocks) [reply gooddata-java/src/main/java/com/gooddata/sdk/service/project/ProjectService.java]: ahh my mistake, reverted in fixup

File: gooddata-java/src/main/java/com/gooddata/sdk/service/project/ProjectService.java (hunk 1 above)
Comment (jimirocks) [reply gooddata-java/src/main/java/com/gooddata/sdk/service/project/ProjectService.java]: done - see the fixup

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-java-merge-compliance-check/132/ : SUCCESS in 15s

**Pull request merged**